src/qengine/opencl.cpp

```diff
@@ -331,18 +331,17 @@ void QEngineOCL::SetDevice(const int& dID, const bool& forceReInit)
     size_t nrmVecAlignSize =
         ((sizeof(real1) * nrmGroupCount) < ALIGN_SIZE) ? ALIGN_SIZE : (sizeof(real1) * nrmGroupCount);
 
-    if (!didInit) {
-#ifdef __APPLE__
-        posix_memalign((void**)&nrmArray, ALIGN_SIZE, nrmVecAlignSize);
-#else
-        nrmArray = (real1*)aligned_alloc(ALIGN_SIZE, nrmVecAlignSize);
-#endif
-    } else if ((oldDeviceID != deviceID) || (nrmGroupCount != oldNrmGroupCount)) {
+	if (didInit && (nrmGroupCount != oldNrmGroupCount)) {
         nrmBuffer = NULL;
-        free(nrmArray);
+        FreeAligned(nrmArray);
         nrmArray = NULL;
-#ifdef __APPLE__
+	}
+
+    if (!didInit || (nrmGroupCount != oldNrmGroupCount)) {
+#if defined(__APPLE__)
         posix_memalign((void**)&nrmArray, ALIGN_SIZE, nrmVecAlignSize);
+#elif defined(_WIN32) && !defined(__CYGWIN__)
```

Comment:
How necessary are the __CYGWIN__ references?  If it's strictly gated on _WIN32, will the right thing happen?

Reply:
> If it's strictly gated on _WIN32, will the right thing happen?

No. _WIN32 will be defined, if building in CYGWIN, but the desired build behavior should be the same as for Linux, in that case.